File: ai/src/main/java/org/pingel/causality/LinearCausalModel.java

package org.pingel.bayes;



public class LinearCausalModel extends CausalModel
{

    public LinearCausalModel() { }
    
    public LinearCausalModel(CausalModel source) {

        // TODO this just does a shallow copy of the source model
        // It should probably to a deep copy.
        
        name = source.name + " (linear)";
        variable2function = source.variable2function;
        newVarIndex = source.newVarIndex;
        name2variable = source.name2variable;
        graph = source.graph;
        
    }
    

}

File: ai/src/main/scala/org/pingel/causality/LinearCausalModel.scala

package org.pingel.bayes

import org.pingel.causality.CausalModel

class LinearCausalModel(source: CausalModel)
  extends CausalModel(source.name + " (linear)") {
  
  // TODO these should be deep copies
  variable2function = source.variable2function
  newVarIndex = source.newVarIndex
  name2variable = source.name2variable
  override val g = source.g
}

